models/memo.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require 'json'
+
+class Memo
+  JSON_PATH = File.expand_path('../memos.json', __dir__)
+
+  attr_reader :id, :title, :content
+
+  class << self
+    def all
+      JSON.load_file(JSON_PATH, symbolize_names: true)
+          .sort_by { |h| h[:id] }
+          .map { |memo_hash| new(**memo_hash) }
+    end
+
+    def find(id)
+      memo_hash = JSON.load_file(JSON_PATH, symbolize_names: true).find { |hash| hash[:id] == id.to_i }
+
+      new(**memo_hash) if memo_hash
```

Comment:
ふわっとした感想なのですが、後置 `if` で書くと「`new` の処理が本流」という感じが薄くなるかなと思いました。あとは、  `memo_hash` が `nil` だったときにこのメソッドが `nil` を返すのを明示したほうがいいかなと思いました（`else` が省略されたときに条件が偽だったら `if` 式は `nil` を返すのは常識、と言われるとちょっと弱いですが）。

```suggestion
      memo_hash ? new(**memo_hash) : nil
```

などとするのはどうでしょうか。